### PR TITLE
[Hotfix] BPTL Dashboard: Update API endpoints to query Shipping Reports (Part 2)

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
Title^^
This PR addresses following issue:
- On load, returns Shipping Report data for last 2 weeks
- Optimized code as needed

Related PR:
https://github.com/episphere/biospecimen/pull/595
https://github.com/episphere/connectFaas/pull/430

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -1572,22 +1572,22 @@ const getSpecimenCollections = async (token, siteCode) => {
...
-        query = query.where('959708259', '==', trackingId);
-    }
-
-    if (endDate) {
-        query = query.where('656548982', '<=', endDate);
+const preQueryBuilder = (filters, query, trackingId, endDate, startDate, source, siteCode) => {
+    if (Object.keys(filters).length === 0 && source === `bptlShippingReport`) {
+        const currentDate = new Date(new Date().getTime()).toISOString();
+        const dateTwoWeeksAgo = new Date(new Date().getTime() - (1209600000)).toISOString();
+        return query = query.where('656548982', '>=', dateTwoWeeksAgo).where('656548982', '<=', currentDate)
```
Comment (JoeArmani) [inline utils/firestore.js:1579]: Can be simplified by just returning `query.where('656548982', '>=', dateTwoWeeksAgo).where('656548982', '<=', currentDate)`

File: utils/firestore.js [hunk 2]
```diff
@@ -1572,22 +1572,22 @@ const getSpecimenCollections = async (token, siteCode) => {
...
     }
-
-    if (startDate) {
-        query = query.where('656548982', '>=', startDate);
+    else {
+        return query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
```
Comment (JoeArmani) [inline utils/firestore.js:1582]: Same thing here

**Review by JoeArmani — APPROVED** [review]
looks good, left minor comments.

**Comment by ramavavi** [comment]
Looks good for me.

**Review by ramavavi — APPROVED** [review]

File: utils/firestore.js [hunk 3]
```diff
@@ -1572,22 +1572,22 @@ const getSpecimenCollections = async (token, siteCode) => {
...
+}
 
-    if (source !== 'bptlShippingReport') {
-        query = query.where('789843387', '==', siteCode);
-    }
+const buildQueryWithFilters = (query, trackingId, endDate, startDate, source, siteCode) => {
+    if (trackingId) query = query.where('959708259', '==', trackingId);
+    if (endDate) query = query.where('656548982', '<=', endDate);
+    if (startDate) query = query.where('656548982', '>=', startDate);
+    if (source !== `bptlShippingReport`) query = query.where('789843387', '==', siteCode);
```
Comment (Gbarra9) [inline utils/firestore.js:1590]: I like the condensed if block statements!

**Review by anthonypetersen — APPROVED** [review]
looks good

**Pull request merged**